lib/daemon_runner/session.rb

```diff
@@ -107,6 +114,18 @@ def destroy!
       Diplomat::Session.destroy(id)
     end
 
+    # Verify wheather the session exists after a period of time
+    def verify_session
+      logger.info(" - Wait until Consul session #{id} exists")
+      Timeout.timeout(2) do
+        sleep 1 if session_exist?
```

Comment:
If your goal is to determine whether the given session actually exists then:

1. 2 seconds is a pretty short period of time considering that calling `Diplomat::Session.list` wraps an HTTP request and could take an indeterminate amount of time.
2. expecting a timeout and rescuing it seems kind of, well, gross. (nb. Is this considered an anti-pattern in Ruby? I know in Python it's completely acceptable to use exceptions for flow control but I'll admit I'm not sure if Ruby is the same way.)